### PR TITLE
MBS-7734,8365,8390,9128,9159,9181: Improve URL auto-select, clean-up and display

### PR DESCRIPTION
**Pull request by y-van-z**
This addresses six URL clean-up issues:
1. [MBS-8390 - Add autoselect + sidebar for Bandsintown](https://tickets.metabrainz.org/browse/MBS-8390)
  Additionally normalize URLs (to `https` scheme and lowercase)
  And validate clean URLs with entity type (`venue` → Place, `event` → Event, no prefix → Artist)
2. [MBS-7734 - Improve Facebook URL clean up](https://tickets.metabrainz.org/browse/MBS-7734)
  Remove 5 tab path suffixes and 16 referral or display related query fields
3. [MBS-8365 - Add autoselect for Pinterest URLs](https://tickets.metabrainz.org/browse/MBS-8365)
  Additionally normalize URLs (to start with `https://www.`) and removes 5 tab path suffixes
4. [MBS-9181 - Improve ReverbNation URL cleanup](https://tickets.metabrainz.org/browse/MBS-9181)
  Remove 7 query fields related to either referral or display
5. [MBS-9159 - Enable URL cleanup for Instagram URLs](https://tickets.metabrainz.org/browse/MBS-9159)
  Remove query and fragment, since they only contain display, language or referral info
6. [MBS-9128 - Remove "list" URL argument in YouTube URLs](https://tickets.metabrainz.org/browse/MBS-9128)
  Actually just fix consistency of regular expressions in clean-up function

Tests have been updated accordingly.

### CODE REVIEW

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -728,7 +731,7 @@ const CLEANUPS = {
     clean: function (url) {
       url = url.replace(/^(https?:\/\/)?([^\/]+\.)?facebook\.com(\/#!)?/, "https://www.facebook.com");
       // Remove ref (where the user came from), sk (subpages in a page, since we want the main link) and a couple others
-      url = url.replace(new RegExp("([&?])(sk|ref|fref|sid_reminder|ref_dashboard_filter)=([^?&]*)", "g"), "$1");
+      url = url.replace(new RegExp("([&?])(__tn__|_fb_noscript|_rdr|acontext|em|entry_point|filter|focus_composer|fref|hc_location|pnref|qsefr|ref|ref_dashboard_filter|ref_type|refsrc|rf|sid_reminder|sk|tab|viewas)=([^?&]*)", "g"), "$1");
```
Comment (Freso) [inline root/static/scripts/edit/URLCleanup.js]: Are there any URL parameters we want to keep at all? Can't we just remove everything after and including a `?`?

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -748,6 +752,40 @@ const CLEANUPS = {
       return true;
     },
   },
+  instagram: {
+    match: [new RegExp("^(https?://)?([^/]+\\.)?instagram\\.com/", "i")],
+    type: LINK_TYPES.socialnetwork,
+    clean: function (url) {
+      var m = url.match(/^(https?:\/\/)?((?:[^\/]+\.)?instagram\.com)\/([^?#]+)(?:[?#].*)?$/);
+      if (m) {
+        var scheme = m[1] || 'https://';
```
Comment (Freso) [inline root/static/scripts/edit/URLCleanup.js]: Don't we just always want to use https here?

File: root/static/scripts/edit/URLCleanup.js (hunk 1 above)
Comment (y-van-z) [inline root/static/scripts/edit/URLCleanup.js]: Short answer: Not as simple, alas. [Long answer on the tickets tracker](https://tickets.metabrainz.org/browse/MBS-7734?focusedCommentId=41803#comment-41803), so to keep other followers of MBS-7734 involved into the discussion.

File: root/static/scripts/edit/URLCleanup.js (hunk 2 above)
Comment (y-van-z) [inline root/static/scripts/edit/URLCleanup.js]: Yes, will do, thanks. Follow-up in a [comment to MBS-9159](https://tickets.metabrainz.org/browse/MBS-9159?focusedCommentId=41804#comment-41804).

Comment (y-van-z) [inline root/static/scripts/edit/URLCleanup.js]: Added in https://github.com/metabrainz/musicbrainz-server/pull/441/commits/56b90d72ccc4a4654708bcbf1fd703128b468ded and https://github.com/metabrainz/musicbrainz-server/pull/441/commits/212e60c220c5270045f9dfe0d4ee2229927de11b (to be squashed after review).

**Pull request merged**